osu.Game/Screens/Edit/EditorTimestampParser.cs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text.RegularExpressions;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Objects.Types;
+
+namespace osu.Game.Screens.Edit
+{
+    public static class EditorTimestampParser
+    {
+        private static readonly Regex timestamp_regex = new Regex(@"^(\d+:\d+:\d+)(?: \((\d+(?:[|,]\d+)*)\))?$", RegexOptions.Compiled);
```

Comment:
A timestamp regex already exists in message formatter:

https://github.com/ppy/osu/blob/master/osu.Game/Online/Chat/MessageFormatter.cs#L44-L46

Reply:
Oh nice, can we make this public or do I just copy it?
I'd rather use the existing one just in case something changes in the future

Reply:
If anything, I'd probably move that one into this static class and reference it here instead.

Reply:
Moving `time_regex` in here, making it public, and referencing it in `MessageFormatter`, surely I understood correctly